tensorly/decomposition/_parafac2.py

```diff
@@ -348,27 +365,59 @@ def parafac_updates(X, w, f):
         factors = parafac_updates(projected_tensor, weights, factors)
 
         if normalize_factors:
-            new_factors = []
-            for factor in factors:
-                norms = T.norm(factor, axis=0)
-                norms = tl.where(
-                    tl.abs(norms) <= tl.eps(factor.dtype),
-                    tl.ones(tl.shape(norms), **tl.context(factors[0])),
-                    norms,
-                )
+            weights, factors = cp_normalize((weights, factors))
+
+        # Start line search if requested.
+        if line_iter:
+            jump = iteration ** (1.0 / acc_pow)
+
+            ls_weights = weights_last + (weights - weights_last) * jump
+            ls_factors = [
+                factors_last[ii] + (factors[ii] - factors_last[ii]) * jump
+                for ii in range(3)
+            ]
```

Comment:
```suggestion
            ls_factors = [
                factors_last[ii] + (factors[ii] - factors_last[ii]) * jump
                for ii, _ in enumerate(factors)
            ]
```
Maybe use enumerate so it's clear that this happens with all factor matrices?